tree/treeplayer/test/dataframe/datasource_trivial.cxx

```diff
@@ -98,3 +98,39 @@ TEST(TTrivialDS, FromATDFWithJitting)
    EXPECT_DOUBLE_EQ(9., *max);
    EXPECT_DOUBLE_EQ(22., *min);
 }
+
+// NOW MT!-------------
+
+TEST(TTrivialDS, FromATDFMT)
+{
+   ROOT::EnableImplicitMT(4);
+   std::unique_ptr<TDataSource> tds(new TTrivialDS(320));
+   TDataFrame tdf(std::move(tds));
```

Comment:
I don't understand why not create the tdf as `TDataFrame tdf(TTrivialDS(320));` Any underlying reason?

Reply:
`unique_ptr` prevents users from calling methods on a TDS that is being used by TDF, making the interface easier to use right (another minor advantage is that it also lets us support non-copyable TDataSources).

Note that 
```
TDataFrame tdf(TTrivialDS(320));
```
wouldn't work but
```
TDataFrame tdf(new TTrivialDS(320));
```
would, so we're not imposing a lot of extra characters.

Reply:
Side note: we need to have this supported in PyROOT too. The current move is great but we need to "diplomatically" move away from it :-)

Reply:
...I was wrong, users cannot in fact write `TDataFrame tdf(new TTrivialDS(320));` as `unique_ptr`'s ctor overload that takes a raw pointer is marked explicit